src/ima.js

```diff
@@ -1046,6 +1052,8 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
     this._contentComplete = true;
     if (this._currentAd && !this._currentAd.isLinear()) {
       this.reset();
+    } else if (!this._hasPostRoll() || this._adError) {
```

Comment:
@yairans post-roll can't fail after playback ended?

Reply:
if post-roll failed, _hasPostRoll wouldn't return false anyway?

Reply:
1. I'll try to make it happen, it's not trivial because the post-roll preloaded
2. No

Reply:
I was able to make the post-roll failure after the playback ended (by seeking to the edge). it's well handled by the ERROR event.